modules/repository/license.go

```diff
@@ -111,3 +183,99 @@ func getLicensePlaceholder(name string) *licensePlaceholder {
 	}
 	return ret
 }
+
+// UpdateRepoLicenses will update repository licenses col if license file exists
+func UpdateRepoLicensesByGitRepo(ctx context.Context, repo *repo_model.Repository, gitRepo *git.Repository) error {
+	if gitRepo == nil {
+		var err error
+		gitRepo, err = git.OpenRepository(ctx, repo.RepoPath())
```

Comment:
I think there is a memory leak here
```suggestion
		gitRepo, err = git.OpenRepository(ctx, repo.RepoPath())
		defer gitRepo.close()
```